src/components/Auth/Auth.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { css } from "@emotion/css";
+import { useTheme2 } from "@grafana/ui";
+import { AuthMethod, CustomMethod, CustomMethodId } from "./types";
+import { AuthMethodSettings } from "./auth-method/AuthMethodSettings";
+import { TLSSettings, Props as TLSSettingsProps } from "./tls/TLSSettings";
+import { Props as BasicAuthProps } from "./auth-method/BasicAuth";
+import {
+  CustomHeaders,
+  Props as CustomHeadersProps,
+} from "./custom-headers/CustomHeaders";
+
+export type Props = {
+  selectedMethod: AuthMethod | CustomMethodId;
+  mostCommonMethod?: AuthMethod | CustomMethodId;
+  visibleMethods?: (AuthMethod | CustomMethodId)[];
+  customMethods?: CustomMethod[];
+  onAuthMethodSelect: (authType: AuthMethod | CustomMethodId) => void;
+  basicAuth?: BasicAuthProps;
+  TLS?: TLSSettingsProps;
+  customHeaders?: CustomHeadersProps;
+};
+
+export const Auth: React.FC<Props> = ({
+  selectedMethod,
+  mostCommonMethod,
+  visibleMethods: visibleMethodsFromProps,
+  customMethods,
+  onAuthMethodSelect,
+  basicAuth,
+  TLS,
+  customHeaders,
+}) => {
+  const { colors } = useTheme2();
+
+  const visibleMethods: (
+    | AuthMethod
+    | CustomMethodId
+  )[] = visibleMethodsFromProps ?? [
+    AuthMethod.BasicAuth,
+    AuthMethod.CrossSiteCredentials,
+    AuthMethod.OAuthForward,
+    AuthMethod.NoAuth,
+    ...(customMethods?.map((m) => m.id) ?? []),
+  ];
+
+  const styles = {
+    container: css({
+      margin: "24px 0",
+      maxWidth: 578,
+    }),
+    selectedMethodFields: css({
+      marginTop: 12,
+    }),
+    authMethods: css({
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "flex-start",
+      padding: "16px 16px 12px",
+      border: `1px solid ${colors.background.secondary}`,
+    }),
+  };
+
+  return (
+    <div className={styles.container}>
```

Comment:
Historically the `DataSourceHttpSettings` component has been bothersome because it didn't give devs the flexibility that composing the individual components can give. Currently in this PR only the Auth component and its props are exposed for plugin developers. Are there use cases for plugin developers to need to render a component between `AuthMethodSettings` and `TLS` or only use the `TLS` or `CustomHeaders` component?

Maybe offering these components up individually for devs to compose would make this more reusable in the long term and also make it easier for devs to focus on the individual APIs for each component.

Reply:
Why do you need to pass one more auth as headers if headers are rendered alongside auth methods? Is it possible that some datasource will have headers as the only auth option? If no, I think it's fine to keep as is.

Reply:
In the datasource settings I see a different picture: https://g94-beta.grafana.net/datasources/edit/P693D0419BB394192. I don't think you'll just use custom headers for datadog, but rather will use some custom auth method, right, even though they might be headers under the hood.

Reply:
Let's leave this for now and if in the future plugin devs need the components separately or without the container styles we can expose them then.